### ISSUE TITLE
Whitelisting scala.collection.generic.ParSetFactory crashes the compiler

### ISSUE THREAD

**Issue by nicolasstucki** [lampepfl/dotty#1808]
It crashes at
```scala
at scala.Predef$.assert(Predef.scala:151)
at dotty.tools.dotc.core.TypeComparer.dotty$tools$dotc$core$TypeComparer$$monitoredIsSubType(TypeComparer.scala:141)
at dotty.tools.dotc.core.TypeComparer$$anonfun$isSubType$1.apply$mcZ$sp(TypeComparer.scala:113)
at dotty.tools.dotc.core.TypeComparer$$anonfun$isSubType$1.apply(TypeComparer.scala:104)
at dotty.tools.dotc.core.TypeComparer$$anonfun$isSubType$1.apply(TypeComparer.scala:104)
at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:136)
```

**Comment by odersky** [comment]
@nicolasstucki:
The failure is triggered by a -Yno-deep-subtypes assertion. Those failures could be real or false negatives. Can you try to run the whitelist test without -Yno-deep-subtypes? Do you get a stackoverflow then? Then it's a real failure. If not, it could be just a really deep subtype check chain, or else an anaomaly caught by the cyclicity check. In any case it would be good to see the complete output what's printed on the failure.

**Comment by nicolasstucki** [comment]
If I remove the `-Yno-deep-subtypes` it compiles and the tests pass.

You can find the full output (with `-Yno-deep-subtypes`) in https://gist.github.com/nicolasstucki/0b2c4758f84a575032bdf50c4e748b6b#file-scala-collection-generic-parsetfactory-md

**Comment by odersky** [comment]
Removed the itype:bug label as it's not confirmed this is a bug.

**Comment by nicolasstucki** [comment]
Fixed with #3796
